src/System.Numerics.Vectors/src/System/Numerics/Vector.cs

```diff
@@ -363,7 +363,8 @@ public unsafe Vector(T[] values, int index)
         {
             if (values == null)
             {
-                throw new ArgumentNullException("values");
+                // Match the JIT's exception type here. For perf, a NullReference is thrown instead of an ArgumentNull.
+                throw new NullReferenceException(SR.Arg_NullArgumentNullRef);
             }
             if (index < 0 || (values.Length - index) < Count)
```

Comment:
I wonder whether the explicit check is even needed. 
Considering the perf is important, could you just do "int dummy = values.Length" and let the hardware throw NRE?